1-js/03-code-quality/06-polyfills/article.md

```diff
@@ -1,92 +1,92 @@
-
 # Polyfills and transpilers
 
-The JavaScript language steadily evolves. New proposals to the language appear regularly, they are analyzed and, if considered worthy, are appended to the list at <https://tc39.github.io/ecma262/> and then progress to the [specification](http://www.ecma-international.org/publications/standards/Ecma-262.htm).
+زبان جاوااسکریپت پیوسته در حال تکامل است. پیشنهادهایی برای بهتر شدن آن به‌طور منظم صورت می‌گیرد، این پیشنهاد‌ها بررسی می‌شوند و اگر ارزشمند باشند، به لیست <https://tc39.github.io/ecma262/> اضافه می‌شوند و سپس به [specification](http://www.ecma-international.org/publications/standards/Ecma-262.htm) راه می‌یابند.
 
-Teams behind JavaScript engines have their own ideas about what to implement first. They may decide to implement proposals that are in draft and postpone things that are already in the spec, because they are less interesting or just harder to do.
+تیم‌های مسئول موتورهای جاوااسکریپت تصمیم می‌گیرند کدام یک را اول پیاده‌سازی کنند. ممکن است تصمیم بگیرند پیشنهادهایی که هنوز به‌صورت پیش‌نویس هستند را اول پیاده‌سازی کنند و چیزهایی که در spec هستند را به بعدتر موکول کنند. دلیلش هم می‌تواند این باشد که شاید بعضی جذابیت کمتری دارند یا انجام آن‌ها سخت‌تر است.
 
-So it's quite common for an engine to implement only the part of the standard.
+پس کاملا طبیعی است که یک موتور فقط بخشی از یک استاندارد را پیاده‌سازی کند.
 
-A good page to see the current state of support for language features is <https://kangax.github.io/compat-table/es6/> (it's big, we have a lot to study yet).
+یک صفحه‌ی خوب برای این که ببینید در حال حاضر چه چیزهایی پشتیبانی می‌شود اینجاست <https://kangax.github.io/compat-table/es6/> (خیلی بزرگ است، ما چیزهای زیادی برای مطالعه داریم).
 
-As programmers, we'd like to use most recent features. The more good stuff - the better!
+به عنوان برنامه‌نویس، ما همیشه دوست داریم از ویژگی‌ها و امکانات جدید استفاده کنیم. هر چه جدیدتر، بهتر!
 
-On the other hand, how to make our modern code work on older engines that don't understand recent features yet?
+از طرف دیگر، دوست داریم کدهایمان روی مرورگرهای قدیمی‌تر که امکانات جدید را نمی‌فهمند هم کار کنند. اما چطور می‌شود این کار را انجام داد؟
 
-There are two tools for that:
+دو ابزار برای این کار وجود دارد:
 
 1. Transpilers.
 2. Polyfills.
```

Comment:
Should be translated here as well, alternatively you can keep their English equivalent in parentheses.